lib/misc.py

```diff
@@ -56,16 +56,21 @@ def build_cache(cli):
    bu.cache.summary_print()
 
 def delete(cli):
-   delete_ct = 0
-   for img in im.Image.glob(cli.image_ref):
-      img.unpack_delete()
-      delete_ct += 1
-   for img in im.Image.glob(cli.image_ref + "_stage[0-9]*"):
-      img.unpack_delete()
-      delete_ct += 1
-   if (delete_ct == 0):
-      ch.FATAL("no image matching glob, can’t delete: %s" % cli.image_ref)
+   fail = False
+   for ref in cli.image_ref:
+      delete_ct = 0
+      for img in im.Image.glob(ref):
+         img.unpack_delete()
+         delete_ct += 1
+      for img in im.Image.glob(ref + "_stage[0-9]*"):
+         img.unpack_delete()
+         delete_ct += 1
+      if (delete_ct == 0):
+         fail = True
+         ch.ERROR("no matching image, can’t delete: %s" % ref)
    bu.cache.worktrees_fix()
+   if (fail):
+      ch.FATAL("one or more invalid images")
```

Comment:
Probably friendlier to keep a count rather than saying “one or more”, and the code complexity is roughly the same.